website/content/docs/external_plugins/_index.md

```diff
@@ -71,7 +71,7 @@ You can include additional (optional) keys in the printed JSON object. These key
 
 * `methods`. This is an array specifying the list of methods, enumerated below, that can be called directly on the plugin entry. The plugin root must always include and implement the `list` method.
 * `cache_ttls`. This specifies how many seconds each method's result should be cached (`ttl` is short for time to live). Currently, Wash caches the result of `list`, `read`, and `metadata`.
-* `attributes`. This represents the entry's attributes (see the [`Attributes/Metadata`](/wash/docs#attributes-metadata) section). Time attributes are specified in Unix seconds. Octal modes must be prefixed with the `0` delimiter (e.g. like `0777`). Hexadecimal modes must be prefixed with the `0x` delimiter (e.g. like `0xabcd`).
+* `attributes`. This represents the entry's attributes (see the [`Attributes/Metadata`](/wash/docs#attributes-metadata) section). Time attributes are specified in Unix seconds. Octal modes must be prefixed with the `0` delimiter (e.g. like `0777`). Hexadecimal modes must be prefixed with the `0x` delimiter (e.g. like `0xabcd`). For entries that can be `read`, provide the size if you know it; otherwise Wash will provide a functional default and update the size when the entry has been `read`.
```

Comment:
Since there's some core plugin stuff that also doesn't provide a size due to download being expensive, I think we should move this bit to the general "plugin concepts" doc. I'd also add a note that the caveat with not providing a size is that you cannot filter those entries on the size attribute using `wash find`.

Reply:
Sorry, that you cannot always filter those entries on the size attribute ("not always" being when you've never read it or when a previous read expired).